### PR TITLE
Defining achievements ability

### PR DESCRIPTION
**Pull request by allenwq**

### CODE REVIEW

File: app/controllers/course/condition/levels_controller.rb [hunk 1]
```diff
@@ -1,12 +1,16 @@
 class Course::Condition::LevelsController < Course::ConditionsController
-  load_and_authorize_resource :level_condition, class: Course::Condition::Level.name, parent: false
+  load_resource :level_condition, class: Course::Condition::Level.name, parent: false
+  authorize_resource :level_condition, class: Course::Condition::Level.name, except: [:new, :create]
 
   def new
+    @level_condition.course = current_course
+    authorize!(:new, @level_condition)
```
Comment (allenwq) [inline app/controllers/course/condition/levels_controller.rb]: @lowjoel There is no course after resource was load. My solution is to manually authorize here after course was set.

File: app/controllers/course/condition/achievements_controller.rb [hunk 2]
```diff
@@ -1,13 +1,17 @@
 class Course::Condition::AchievementsController < Course::ConditionsController
-  load_and_authorize_resource :achievement_condition,
-                              class: Course::Condition::Achievement.name, parent: false
+  load_resource :achievement_condition, class: Course::Condition::Achievement.name, parent: false
+  authorize_resource :achievement_condition, class: Course::Condition::Achievement.name,
```
Comment (lowjoel) [inline app/controllers/course/condition/achievements_controller.rb]: you can add a before_action to set the `@achievement_condition.course` before authorize_resource, then you won't need to manually authorize.

**Comment by lowjoel** [comment]
Any ability specs?

File: app/controllers/course/condition/levels_controller.rb (hunk 1 above)
Comment (lowjoel) [inline app/controllers/course/condition/levels_controller.rb]: See comment above

File: app/controllers/course/condition/achievements_controller.rb (hunk 2 above)
Comment (allenwq) [inline app/controllers/course/condition/achievements_controller.rb]: OK ! Thanks

**Comment by allenwq** [comment]
@lowjoel Yah, ability specs are there, not enough ?

**Comment by allenwq** [comment]
@lowjoel Done.

**Pull request merged**